cloud/cluster/memory/subscription.go

```diff
@@ -0,0 +1,56 @@
+package memory
+
+import (
+	"github.com/scootdev/scoot/cloud/cluster"
+)
+
+// subscriber promptly reads its input and maintains a queue so that the sender
+// doesn't have to worry about a send blocking.
+type subscriber struct {
+	inCh    chan []cluster.NodeUpdate
+	outCh   chan []cluster.NodeUpdate
+	closeCh chan interface{}
+	queue   []cluster.NodeUpdate
+}
+
+func makeSubscription(initial []cluster.Node, closeCh chan interface{}, inCh chan []cluster.NodeUpdate) cluster.Subscription {
+	s := &subscriber{
+		inCh:    inCh,
+		outCh:   make(chan []cluster.NodeUpdate),
+		closeCh: closeCh,
+		queue:   nil,
+	}
+	go s.loop()
```

Comment:
Yeah, in java, it would be a little weird to not mention.

I think it is different in go, and so I'm pushing myself to try to adapt to this style: goroutines are so cheap that we don't worry about them.